crates/experimentation_client/src/interface.rs

```diff
@@ -187,9 +187,11 @@ pub extern "C" fn get_applicable_variant(
     let variants = local.block_on(&Runtime::new().unwrap(), unsafe {
         (*client).get_applicable_variant(&context, toss as i8)
     });
-    // println!("variantIds: {:?}", variants);
-    match serde_json::to_string::<Vec<String>>(&variants) {
-        Ok(result) => rstring_to_cstring(result).into_raw(),
+    match variants {
```

Comment:
Can we use map() on variants so that it avoids this match statement? map_err for the error and you can end with ?

Reply:
This function does not return result type.

Reply:
How does the match statement work?